recipes/md4c/all/conanfile.py

```diff
@@ -22,62 +25,76 @@ class Md4cConan(ConanFile):
+            del self.settings.compiler.libcxx
+        except Exception:
+            pass
+        try:
+            del self.settings.compiler.cppstd
+        except Exception:
+            pass
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def validate(self):
-        if self.settings.os != "Windows" and self.options.encoding == "utf-16":
-            raise tools.ConanInvalidConfiguration("utf-16 options is not supported on non-Windows platforms")
+        if self.info.settings.os != "Windows" and self.info.options.encoding == "utf-16":
+            raise ConanInvalidConfiguration(f"{self.ref} doesn't support utf-16 options on non-Windows platforms")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-            destination=self._source_subfolder, strip_root=True)
-
-    @functools.lru_cache(1)
-    def _configure_cmake(self):
-        cmake = CMake(self)
+        get(self, **self.conan_data["sources"][self.version],
+                  destination=self.source_folder, strip_root=True)
 
+    def generate(self):
+        tc = CMakeToolchain(self)
         if self.options.encoding == "utf-16":
-            cmake.definitions["CONAN_C_FLAGS"] = "-DMD4C_USE_UTF16"
+            tc.cache_variables["CONAN_C_FLAGS"] = "-DMD4C_USE_UTF16"
         elif self.options.encoding == "ascii":
-            cmake.definitions["CONAN_C_FLAGS"] = "-DMD4C_USE_ASCII"
-
-        cmake.configure()
-        return cmake
+            tc.cache_variables["CONAN_C_FLAGS"] = "-DMD4C_USE_ASCII"
+        tc.generate()
 
     def build(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-
-        cmake = self._configure_cmake()
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
     def package(self):
-        self.copy(pattern="LICENSE.md", dst="licenses", src=self._source_subfolder)
-        cmake = self._configure_cmake()
+        copy(self, pattern="LICENSE.md", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        cmake = CMake(self)
         cmake.install()
-        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
 
     def package_info(self):
         self.cpp_info.libs = ["md4c", "md4c-html",]
 
         if self.options.encoding == "utf-16":
             self.cpp_info.defines.append("MD4C_USE_UTF16")
+        elif self.options.encoding == "ascii":
+            self.cpp_info.defines.append("MD4C_USE_ASCII")
+
+        self.cpp_info.set_property("cmake_file_name", "md4c")
+        self.cpp_info.set_property("cmake_target_name", "md4c::md4c")
+        self.cpp_info.set_property("pkg_config_name", "md4c")
```

Comment:
There are 2 different targets and pc files, one for each lib: https://github.com/mity/md4c/blob/release-0.4.8/src/CMakeLists.txt#L34-L55

Moreover md4c-html depends on md4c, not the opposite.